pkg/parquetquery/nilIter.go

```diff
@@ -0,0 +1,201 @@
+package parquetquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	pq "github.com/parquet-go/parquet-go"
+)
+
+// NilAttributeIterator copies all functions of the sync iterator with just the next() function being different
+type NilAttributeIterator struct {
+	syncIterator          SyncIterator
+	lastRowNumberReturned RowNumber
+	attrFound             bool
+	lastBuff              bool
+}
+
+var _ Iterator = (*NilAttributeIterator)(nil)
+
+func NewNilAttributeIterator(ctx context.Context, rgs []pq.RowGroup, column int, columnName string, readSize int, filter Predicate, selectAs string, maxDefinitionLevel int, opts ...SyncIteratorOpt) *NilAttributeIterator {
+	// Create the sync iterator
+	syncIterator := NewSyncIterator(ctx, rgs, column, opts...)
+	// Apply options
+	for _, opt := range opts {
```

Comment:
This should not be necessary as the options are already applied in `NewSyncIterator`, right?